dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +123,225 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    def set_up_state_api(self):
+        """
+        Set up the state API in order to fetch task information.
+        """
+        gcs_channel = self._dashboard_head.aiogrpc_gcs_channel
+        self._state_api_data_source_client = StateDataSourceClient(
+            gcs_channel, self._dashboard_head.gcs_aio_client
+        )
+        self._state_api = StateAPIManager(self._state_api_data_source_client)
+
+    async def get_worker_details_for_task(self, task_id, attempt_number):
+        """
+        Retrieves worker details for a specific task and attempt number.
+
+        Args:
+            task_id (str): The ID of the task.
+            attempt_number (int): The attempt number of the task.
+
+        Returns:
+            tuple: A tuple containing the worker's PID (process ID) and IP address.
+
+        Raises:
+            ValueError: If the PID or IP is None, indicating that the information could not be fetched.
+            ValueError: If the task attempt is not in the "RUNNING" state.
+        """
+        option = ListApiOptions(
+            filters=[
+                ("task_id", "=", task_id),
+                ("attempt_number", "=", attempt_number),
+            ],
+            detail=True,
+            timeout=10,
+        )
+
+        result = await self._state_api.list_tasks(option=option)
+        tasks = result.result
+
+        pid = tasks[0]["worker_pid"]
+        logger.info(f"pid {type(pid)}: {pid}")
```

Comment:
can we remove all the debug log?

Reply:
Sure, I removed those logs locally, but my bad, I did not update the latest commit to the upstream.